tools/runners/run-mongo.js

```diff
@@ -584,35 +584,41 @@ var launchMongo = function (options) {
         });
       }
 
-      var initiateResult = yieldingMethod(
-        db.admin(), 'command', {replSetInitiate: configuration});
+      var initiateResult;
+      try{
+        initiateResult = yieldingMethod(
+          db.admin(), 'command', {replSetInitiate: configuration});
+      } catch (e){
+        if(e.message !== 'already initialized') {
+          throw Error("rs.initiate error: " + e.message);
+        }
+      }
+
       if (stopped) {
         return;
       }
-      // why this isn't in the error is unclear.
-      if (initiateResult && initiateResult.documents
-          && initiateResult.documents[0]
-          && initiateResult.documents[0].errmsg) {
-        var err = initiateResult.documents[0].errmsg;
-        if (err !== "already initialized") {
-          throw Error("rs.initiate error: " +
-                      initiateResult.documents[0].errmsg);
-        }
-      }
+
       // XXX timeout eventually?
       while (!stopped) {
-        var status = yieldingMethod(db.admin(), 'command',
-                                    {replSetGetStatus: 1});
-        if (!(status && status.documents && status.documents[0])) {
+        var status;
+        try{
+          var status = yieldingMethod(db.admin(), 'command',
+                                      {replSetGetStatus: 1});
+        } catch (e) {
+          //Some of the expected results, come as errors.
+          status = e;
```

Comment:
Does the `startupStatus === 6` error come through as an error or a return value? If the second, I think we should just let the original error through here, rather than catching and then throwing again later.